corehq/apps/reports/views.py

```diff
@@ -1771,6 +1771,14 @@ def _error(msg):
             non_parents = filter(lambda cb: cb.path == [], case_blocks)
             if len(non_parents) == 1:
                 edit_session_data['case_id'] = non_parents[0].caseblock.get(const.CASE_ATTR_ID)
+                case = CaseAccessors(domain).get_case(edit_session_data['case_id'])
+                if case.closed:
+                    return _error(_(
+                        u'Case {} is closed. Please reopen the '
+                        u'case before editing the form'
+                    ).format(case.case_id))
+                elif case.deleted:
+                    return _error(_(u'Case {} is deleted. Cannot edit this form.').format(case.case_id))
```

Comment:
Should this not also apply to any of the cases edited by the form?

Also would be nice to use have the error link to the case using the case name instead of just have the case ID.